File: feed/src/main/java/com/example/feed/FeedComponent.java
package com.example.feed;

import com.example.core.ActivityScope;
import com.example.core.CoreComponent;
import dagger.Component;

@ActivityScope
@Component(
        dependencies = CoreComponent.class,
        modules = FeedModule.class
)
public interface FeedComponent {

    void inject(FeedActivity activity);

}

File: feed/src/main/java/com/example/feed/FeedComponent.kt
package com.example.feed

import com.example.core.ActivityScope
import com.example.core.CoreComponent
import dagger.Component

@ActivityScope
@Component(dependencies = arrayOf(CoreComponent::class), modules = arrayOf(FeedModule::class))
interface FeedComponent {

    fun inject(activity: FeedActivity)

}
